x-pack/plugins/ml/public/alerting/ml_alerting_flyout.tsx

```diff
@@ -55,15 +55,18 @@ export const MlAnomalyAlertFlyout: FC<MlAnomalyAlertFlyoutProps> = ({
     if (initialAlert) {
       return triggersActionsUi.getEditAlertFlyout({
         ...commonProps,
-        initialAlert,
+        initialRule: {
+          ...initialAlert,
+          ruleTypeId: initialAlert.alertTypeId,
+        },
```

Comment:
what if it's already contained `ruleTypeId`?

Reply:
I see, I will add more logic

Reply:
Maybe this:
```
diff --git a/x-pack/plugins/ml/public/alerting/ml_alerting_flyout.tsx b/x-pack/plugins/ml/public/alerting/ml_alerting_flyout.tsx
index 84e1e63daed..850c3dbada7 100644
--- a/x-pack/plugins/ml/public/alerting/ml_alerting_flyout.tsx
+++ b/x-pack/plugins/ml/public/alerting/ml_alerting_flyout.tsx
@@ -12,9 +12,10 @@ import { useMlKibana } from '../application/contexts/kibana';
 import { ML_ALERT_TYPES } from '../../common/constants/alerts';
 import { PLUGIN_ID } from '../../common/constants/app';
 import { MlAnomalyDetectionAlertRule } from '../../common/types/alerts';
+import { Rule } from '../../..//triggers_actions_ui/public';
 
 interface MlAnomalyAlertFlyoutProps {
-  initialAlert?: MlAnomalyDetectionAlertRule;
+  initialAlert?: MlAnomalyDetectionAlertRule & Rule;
   jobIds?: JobId[];
   onSave?: () => void;
   onCloseFlyout: () => void;
@@ -57,7 +58,7 @@ export const MlAnomalyAlertFlyout: FC<MlAnomalyAlertFlyoutProps> = ({
         ...commonProps,
         initialRule: {
           ...initialAlert,
-          ruleTypeId: initialAlert.alertTypeId,
+          ruleTypeId: initialAlert.ruleTypeId ?? initialAlert.alertTypeId,
         },
       });
     }
@@ -127,7 +128,7 @@ export const JobListMlAnomalyAlertFlyout: FC<JobListMlAnomalyAlertFlyoutProps> =
 };
 
 interface EditRuleFlyoutProps {
-  initialAlert: MlAnomalyDetectionAlertRule;
+  initialAlert: MlAnomalyDetectionAlertRule & Rule;
   onSave: () => void;
 }
 
cbr-mbp:kibana chris$ 
```

Reply:
I think @chrisronline suggestion makes sense